### PR TITLE
docs(readme): add common routing api

### PR DESCRIPTION
**Pull request by JAdshead**

### CODE REVIEW

**Comment by JAdshead** [comment]
What is the communication plan for this change?
Does any documentation need to be updated or added to account for this change? If so was it done already?
What is the motivation for this change?
Should these changes also be applied to a maintenance branch or any other long lived branch?
How was this change tested? NA
Does this change require cross browser checks? Why or why not? NA - docs update
Does this change require a performance test prior to merging? Why or why not? NA- docs update
Could this be considered a breaking change? Why or why not? No, docs update
Does the change impact caching? No
Does the change impact HTTP headers? No
Does the change have any new infrastructure requirements? no
Does the change affect other versions of the app? no 
Does the change require additional environment variables? no
What is the impact to tenants? none
What is the impact to individual users? none
What is the change to the size of assets? no
Should integration tests be added to protect against future regressions on this change? no

File: docs/api/modules/Loading-Modules.md [hunk 1]
```diff
@@ -20,10 +20,17 @@ Both methods are described in the following.
 
 > 👍 Most commonly used method to load Holocron Modules
 
-A parent Module may add the `ModuleRoute` routing component to the [`childRoutes` Module Lifecycle Hook](#routing) to load a child Module dynamically on the server or browser when matching a route path. Once the Module is loaded, it is injected as a JSX element into the `children` prop of the parent Module.
+A parent Module may add the `ModuleRoute` routing component to the [`childRoutes` Module Lifecycle Hook](#./routing.md) to load a child Module dynamically on the server or browser when matching a route path. Once the Module is loaded, it is injected as a JSX element into the `children` prop of the parent Module.
```
Comment (mtomcal) [inline docs/api/modules/Loading-Modules.md]: `#./routing.md` to `./routing.md`

File: docs/api/modules/Routing.md [hunk 2]
```diff
@@ -2,6 +2,76 @@
...
+Please see One App Router's [Link API docs](https://github.com/americanexpress/one-app-router/blob/master/docs/API.md#link) for more information.
+
+## `onEnter`
+
+Both `Route` and `ModuleRoute` accept an `onEnter` hook which is called before the component or module gets
+rendered.
+
+Please see One App Router's [`onEnter` API](https://github.com/americanexpress/one-app-router/blob/master/docs/API.md#onenternextstate-replace-callback)  for more information.
+
+
```
Comment (mtomcal) [inline docs/api/modules/Routing.md]: Extra newline.

File: docs/api/modules/Routing.md [hunk 3]
```diff
@@ -2,6 +2,76 @@
...
+MyModule.onEnterRouteHook = (nextState, replace, callback) => {
+  if (!authorized) {
+    replace('/login');
+  }
+  callback();
+};
+```
+
+See the [Holcoron Module Route docs](https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route/README.md#onEnterRouteHook) for information.
+
```
Comment (mtomcal) [inline docs/api/modules/Routing.md]: Perhaps add a More Information section with linkage to Frank modules and other related APIs or external libraries you can think of.

File: docs/api/modules/Routing.md [hunk 4]
```diff
@@ -2,6 +2,76 @@
...
+**Contents**
+* [Router](#router)
+* [Route](#route)
+* [childRoutes](#childroutes)
+* [Link](#link)
+* [onEnterRouteHook](#onenterroutehook)
+
+**📘 More Information**
+* Loading Holocron Module via Route: [ModuleRoute](./loading-modules.md#moduleroute)
```
Comment (mtomcal) [inline docs/api/modules/Routing.md]: This link is broken:
```./loading-modules.md#moduleroute``` to ```./Loading-Modules.md#moduleroute```

File: docs/api/modules/Routing.md [hunk 5]
```diff
@@ -2,6 +2,76 @@
...
+
```
Comment (mtomcal) [inline docs/api/modules/Routing.md]: Also I would have a header and link-out to `ModuleRoute`. People may come looking for `ModuleRoute` info first here in this doc and not Loading Modules.

File: docs/api/modules/Routing.md [hunk 6]
```diff
@@ -2,6 +2,76 @@
 
 # Routing
 
-Documentation Forthcoming
+One App uses **[@americanexpress/one-app-router](https://github.com/americanexpress/one-app-router)**, a fork of **`react-router@3`**
```
Comment (mtomcal) [inline docs/api/modules/Routing.md]: Link out to `react-router@3`

File: README.md [hunk 7]
```diff
@@ -133,7 +133,75 @@ For a module to act as the root module the only requirements are:
...
+itself is required to have at least a single route with a path.
+
+```js
+import { Route } from '@americanexpress/one-app-router';
+
+export function RootModule({ children, config }) {
+  return <h1>Hello World!</h1>;
+}
+RootModule.childRoutes = [
+  <Route path="/" />,
```
Comment (10xLaCroixDrinker) [inline README.md]: is path the only required prop?

File: docs/api/modules/Loading-Modules.md [hunk 8]
```diff
@@ -20,10 +20,17 @@ Both methods are described in the following.
...
 ### `ModuleRoute`
 
+Similar to One App Router's `<Route>` with an additional prop, `moduleName`, [ModuleRoute](https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route) loads
+a Holocron Module when the URL matches the given path.
+
+```js
+<ModuleRoute path="/home" module-name="my-module" />;
+```
+
 Please see [`ModuleRoute`](https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route#-usage) in the Holocron Module Route API.
```
Comment (10xLaCroixDrinker) [inline docs/api/modules/Loading-Modules.md]: link directly to https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route#moduleroute

File: docs/api/modules/Routing.md [hunk 9]
```diff
@@ -2,6 +2,76 @@
...
+uplifted to work with React@^17, for its client and server side routing.
+
+**Contents**
+* [Router](#router)
+* [Route](#route)
+* [childRoutes](#childroutes)
+* [Link](#link)
+* [onEnterRouteHook](#onenterroutehook)
```
Comment (10xLaCroixDrinker) [inline docs/api/modules/Routing.md]: can we get backticks around the text for these links?

File: docs/api/modules/Routing.md [hunk 10]
```diff
@@ -2,6 +2,76 @@
...
+
+`<Route>` is used to define the mapping between a path and component. For use with One App this must be defined
+within `childRoutes` defined on a module.
+
+Please see One App Router's [Route API docs](https://github.com/americanexpress/one-app-router/blob/master/docs/API.md#route) for more information.
+
+## `childRoutes`
+
+Enables components and modules to define their own child routes. `childRoutes` can be either a single
+route, array of routes or a function which accepts the Redux store as the sole argument.
```
Comment (10xLaCroixDrinker) [inline docs/api/modules/Routing.md]: not e that the function should return a route or array of routes

File: docs/api/modules/Routing.md [hunk 11]
```diff
@@ -2,6 +2,76 @@
...
+```js
+MyModule.onEnterRouteHook = (nextState, replace, callback) => {
+  if (!authorized) {
+    replace('/login');
+  }
+  callback();
+};
+```
+
+See the [Holcoron Module Route docs](https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route/README.md#onEnterRouteHook) for information.
```
Comment (10xLaCroixDrinker) [inline docs/api/modules/Routing.md]: https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route#onenterroutehook

File: README.md (hunk 7 above)
Comment (JAdshead) [reply README.md]: for the router to match and load the child module, yes

File: README.md [hunk 12]
```diff
@@ -133,7 +133,75 @@ For a module to act as the root module the only requirements are:
 
 ### Making an API call
 
-### Adding a Route
+### Routing and Navigation
+
+One App contains a single route `<ModuleRoute moduleName={root-module-name}/>` which loads the
+root-module configured by `root-module-name`. This is what makes the root module the entry
+point to your application. As One App does not set a path for the root module the root module
+itself is required to have at least a single route with a path.
```
Comment (mtomcal) [inline README.md]:
```suggestion
point to your application. As One App does not set a path by default, the root module
is required to have at least a single route with a path.
```

File: README.md [hunk 13]
```diff
@@ -133,7 +133,75 @@ For a module to act as the root module the only requirements are:
...
+
+export function RootModule({ children, config }) {
+  return <h1>Hello World!</h1>;
+}
+RootModule.childRoutes = [
+  <Route path="/" />,
+];
+```
+
+As your application grows you can also take advantage of holocron's [`ModuleRoute`](./docs/api/modules/loading-modules.md#moduleroute)
```
Comment (mtomcal) [inline README.md]:
```suggestion
As your application grows, you can also take advantage of Holocron's [`ModuleRoute`](./docs/api/modules/loading-modules.md#moduleroute)
```

File: docs/api/modules/Loading-Modules.md [hunk 14]
```diff
@@ -20,10 +20,17 @@ Both methods are described in the following.
 
 > 👍 Most commonly used method to load Holocron Modules
 
-A parent Module may add the `ModuleRoute` routing component to the [`childRoutes` Module Lifecycle Hook](#routing) to load a child Module dynamically on the server or browser when matching a route path. Once the Module is loaded, it is injected as a JSX element into the `children` prop of the parent Module.
+A parent Module may add the `ModuleRoute` routing component to the [`childRoutes` Module Lifecycle Hook](#./Routing.md) to load a child Module dynamically on the server or browser when matching a route path. Once the Module is loaded, it is injected as a JSX element into the `children` prop of the parent Module.
```
Comment (mtomcal) [inline docs/api/modules/Loading-Modules.md]:
```suggestion
A parent Module may add the `ModuleRoute` routing component to the [`childRoutes` Module Lifecycle Hook](./Routing.md) to load a child Module dynamically on the server or browser when matching a route path. Once the Module is loaded, it is injected as a JSX element into the `children` prop of the parent Module.
```

File: docs/api/modules/Loading-Modules.md [hunk 15]
```diff
@@ -20,10 +20,17 @@ Both methods are described in the following.
...
 
 ### `ModuleRoute`
 
+Similar to One App Router's `<Route>` with an additional prop, `moduleName`, [ModuleRoute](https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route#moduleroute) loads
```
Comment (mtomcal) [inline docs/api/modules/Loading-Modules.md]:
```suggestion
Similar to One App Router's `<Route>` with an additional prop, `moduleName`, [`ModuleRoute`](https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route#moduleroute) loads
```

File: docs/api/modules/Routing.md [hunk 16]
```diff
@@ -2,6 +2,82 @@
...
+```js
+MyModule.onEnterRouteHook = (nextState, replace, callback) => {
+  if (!authorized) {
+    replace('/login');
+  }
+  callback();
+};
+```
+
+Please see the [Holcoron Module Route docs](https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route#onenterroutehook) for information.
```
Comment (mtomcal) [inline docs/api/modules/Routing.md]:
```suggestion
Please see the [Holocron Module Route docs](https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route#onenterroutehook) for information.
```

File: docs/api/modules/Routing.md [hunk 17]
```diff
@@ -2,6 +2,82 @@
...
+```
+
+Please see the [Holocron Module Route docs](https://github.com/americanexpress/holocron/tree/master/packages/holocron-module-route#onenterroutehook) for information.
+
+## ModuleRoute
+
+Please see [ModuleRoute](./Loading-Modules.md#moduleroute) for more information.
+
+**📘 More Information**
+* For an root module implementation example see [`FrankLloydRoot`](../../../prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/components/FrankLloydRoot.jsx)
```
Comment (mtomcal) [inline docs/api/modules/Routing.md]:
```suggestion
* For a Root Module implementation example, see [`FrankLloydRoot`](../../../prod-sample/sample-modules/frank-lloyd-root/0.0.0/src/components/FrankLloydRoot.jsx)
```

**Pull request merged**